parlai/core/image_featurizers.py

```diff
@@ -60,33 +58,35 @@ class ImageLoader():
     """Extract image feature using pretrained CNN network.
     """
     def __init__(self, opt):
-        self.opt = copy.deepcopy(opt)
-        self.use_cuda = not opt.get('no_cuda', False) and torch.cuda.is_available()
+        self.opt = opt.copy()
+        self.use_cuda = False
         self.netCNN = None
         self.im = opt['image_mode']
         if self.im is not None and self.im not in ['none', 'raw', 'ascii']:
-            self.init_cnn()
+            self.init_cnn(self.opt)
 
-    def init_cnn(self):
+    def init_cnn(self, opt):
         """Lazy initialization of preprocessor model in case we don't need any image preprocessing."""
         try:
             import torch
+            self.use_cuda = (not opt.get('no_cuda', False)
+                             and torch.cuda.is_available())
+            self.torch = torch
         except ModuleNotFoundError:
             raise ModuleNotFoundError('Need to install Pytorch: go to pytorch.org')
         from torch.autograd import Variable
         import torchvision
         import torchvision.transforms as transforms
         import torch.nn as nn
 
-        opt = self.opt
         self.image_size = opt['image_size']
         self.crop_size = opt['image_cropsize']
         self.datatype = opt['datatype']
         self.image_mode = opt['image_mode']
 
         if self.use_cuda:
             print('[ Using CUDA ]')
-            torch.cuda.set_device(opt.get('gpu', 0))
+            torch.cuda.set_device(opt.get('gpu', opt.get('gpu', -1)))
```

Comment:
does this need to be nested like this? or can it just be ``opt.get('gpu', -1)``?